src/compiler/checker.ts

```diff
@@ -4735,6 +4859,1104 @@ namespace ts {
+                                // `var` is `FunctionScopedVariable`, `const` and `let` are `BlockScopedVariable`, and `module.exports.thing =` is `Property`
+                                const flags = !(symbol.flags & SymbolFlags.BlockScopedVariable) ? undefined : isConstVariable(symbol) ? NodeFlags.Const : NodeFlags.Let;
+                                const name = (needsPostExportDefault || !(symbol.flags & SymbolFlags.Property)) ? localName : getUnusedName(localName, symbol);
+                                let textRange: Node | undefined = symbol.declarations && filter(symbol.declarations, d => isVariableDeclaration(d))[0];
+                                if (textRange && isVariableDeclarationList(textRange.parent) && textRange.parent.declarations.length === 1) {
+                                    textRange = textRange.parent.parent;
+                                }
+                                const statement = setTextRange(createVariableStatement(/*modifiers*/ undefined, createVariableDeclarationList([
+                                    createVariableDeclaration(name, serializeTypeForDeclaration(type, symbol))
+                                ], flags)), textRange);
+                                addResult(statement, name !== localName ? modifierFlags & ~ModifierFlags.Export : modifierFlags);
+                                if (name !== localName && !isPrivate) {
+                                    // We rename the variable declaration we generate for Property symbols since they may have a name which conflicts with a local declaration. Eg,
+                                    // `module.exports.g = g` - we want to make a variable declaration with type `typeof g` (ergo, we must emit `g`)
+                                    // and so we can't name the variable declaration `g` (since there's already a local `g`) - so we anonymize the local variable declaration and make
+                                    // an export declaration with an alias to the right name
+                                    addResult(createExportDeclaration(/*decorators*/ undefined, /*modifiers*/ undefined, createNamedExports([createExportSpecifier(name, localName)])), ModifierFlags.None);
+                                }
+                            }
+                        }
+                    }
+                    if (symbol.flags & SymbolFlags.Enum) {
+                        addResult(createEnumDeclaration(
+                            /*decorators*/ undefined,
+                            createModifiersFromModifierFlags(isConstEnumSymbol(symbol) ? ModifierFlags.Const : 0),
+                            getInternalSymbolName(symbol, symbolName),
+                            map(filter(getPropertiesOfType(getTypeOfSymbol(symbol)), p => !!(p.flags & SymbolFlags.EnumMember)), p => {
+                                // TODO: Handle computed names

```

Comment:
I don't think there are at the moment, since the failure can't be witnessed, as js special declarations don't yet support computed names - see the two PRs we have open to add that support, https://github.com/microsoft/TypeScript/pull/33220 and https://github.com/microsoft/TypeScript/pull/33537